### ISSUE TITLE
ASYNC250 false positive for input call wrapped into lambda

### ISSUE THREAD

**Issue by RomanValov** [astral-sh/ruff#23425]
### Summary

Hello,

with the recent updates ruff started to complain on following snippet of code:

```python
import asyncio


async def main() -> None:
    input("sync")
    await asyncio.to_thread(lambda: input("async"))


if __name__ == '__main__':
    asyncio.run(main())
```

```sh
$ ruff check --select=ASYNC250 async.py
ASYNC250 Blocking call to `input()` in async context
 --> async.py:6:5
  |
5 | async def main() -> None:
6 |     input("sync")
  |     ^^^^^
7 |     await asyncio.to_thread(lambda: input("async"))
  |

ASYNC250 Blocking call to `input()` in async context
 --> async.py:7:37
  |
5 | async def main() -> None:
6 |     input("sync")
7 |     await asyncio.to_thread(lambda: input("async"))
  |                                     ^^^^^
  |

Found 2 errors.
```

for comparison flake8-async passes "async" invocation:

```sh
$ flake8 --enable=ASYNC250 async.py
async.py:6:5: ASYNC250 Blocking sync call `input()` in async function. Wrap in `asyncio.loop.run_in_executor`.
```

### Version

ruff 0.15.1

**Comment by ntBre** [comment]
Thanks for the report! Looks like the rule uses this function:

https://github.com/astral-sh/ruff/blob/1425c185b0a47be87112762f65b5bf7e323fb950/crates/ruff_python_semantic/src/model.rs#L1595-L1603

which I think maybe should return `false` if it encounters a `lambda`, more like the version on `Checker`:

https://github.com/astral-sh/ruff/blob/1425c185b0a47be87112762f65b5bf7e323fb950/crates/ruff_linter/src/checkers/ast/mod.rs#L799-L811

cc @amyreese
